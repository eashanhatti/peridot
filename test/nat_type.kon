let {
  datatype Nat : Type {
  	zero : Nat;
  	succ : [_ : Nat] -> Nat;
	};
} in {
	(pure unit)
}
