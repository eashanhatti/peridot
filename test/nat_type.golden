RIGHT{([],
 Let
   [ObjectConstant
      (Id {unId = 0})
      (TypeType (SUniVar (Global {unGlobal = 1}))),
    ObjectConstant
      (Id {unId = 1}) (GlobalVar (Id {unId = 0})),
    ObjectConstant
      (Id {unId = 2})
      (FunType
         Explicit
         (GlobalVar (Id {unId = 0}))
         (GlobalVar (Id {unId = 0})))]
   (IOIntro1 UnitIntro))
}